lib/coin_flip_demo.dart

```diff
@@ -0,0 +1,203 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'dart:math';
+
+final Random random = Random();
+final NumberFormat formatter = NumberFormat.percentPattern('en_US');
+
+class CoinFlipDemo extends StatefulWidget {
+  @override
+  CoinFlipDemoState createState() => CoinFlipDemoState();
+}
+
+class CoinFlipDemoState extends State<CoinFlipDemo> with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+  Animation _flipAnimation;
+
+  // Whether the coin is currently flipping
+  bool isCoinFlipping = false;
+  // The number of times the coin flip resulted in heads.
+  int _headsCount = 0;
+  // The number of times the coin flip resulted in tails.
+  int _tailsCount = 0;
+  // The total number of times the coin has been flipped.
+  int _totalFlipCount = 0;
+
+  void _flipCoin() {
+    if (!isCoinFlipping) {
+      _headsCount = 0;
```

Comment:
probably idiomatic to put these 3 lines inside the setState call as they are part of the visible ui state of the component.

Reply:
I actually don't want the UI updating when I set these values. If we update them, they are there so briefly that it makes the UI flash and look janky. The coin flipping is so quick that it doesn't really make sense to display 0's when the calculation is occurring.